crema/parsers/comet.py

```diff
@@ -0,0 +1,206 @@
+    else:
+        txt_files = utils.listify(txt_files)
+        for txt_file in txt_files:
+            with open(txt_file) as txt_ref:
+                # First line of Comet output consists only of version
+                # If statement below in case first line is removed
+                line = txt_ref.readline().rstrip()
+                if line.startswith("CometVersion"):
+                    line = txt_ref.readline().rstrip()
+                    skip_first_line = True
+                cols = line.split("\t")
+                scores = scores.intersection(set(cols))
+
+    if not scores:
+        raise ValueError(
+            "Could not find any of the Comet score columns in all of the files."
+            f"The columns Crema looks for are {', '.join(list(scores_all))}"
+        )
+
+    scores = list(scores)
+
+    # Read in the files:
+    fields = spectrum + [peptide] + [target] + scores + [pairing] + [protein]
+    if isinstance(txt_files, pd.DataFrame):
+        data = txt_files.copy(deep=copy_data).loc[:, fields]
+    else:
+        data = pd.concat(
+            [_parse_psms(f, fields, skip_first_line) for f in txt_files]
+        )
+
+    data["target/decoy"] = ~data[protein].str.contains("DECOY_")
+
+    psms = read_txt(
+        data,
+        target_column=target,
+        spectrum_columns=spectrum,
+        score_columns=scores,
+        peptide_column=peptide,
+        protein_column=protein,
+        protein_delim=protein_delim,
+        sep="\t",
+        copy_data=False,
+    )
+
+    # Remove first and last amino acid from sequence
+    # Looks like "R.WVNEK.Y"
+    peptide_column = psms.peptides
+    new_peptide_column = peptide_column.str[2:-2]
+    psms.set_peptide_column(new_peptide_column)
+
+    # always pair target and decoys for Comet
+    if pairing_file_name == None:
+        # implicit pairing based off fact that Comet reverses peptides
+        psms._peptide_pairing = _create_pairing(data)
+    else:  # explicit pairing
+        psms._peptide_pairing = utils.create_pairing_from_file(
+            pairing_file_name
+        )
+
+    # Remove decoy prefix from protein ID
+    protein_column = psms.proteins
+    new_protein_column = protein_column.str.replace("DECOY_", "", regex=True)
+    psms.set_protein_column(new_protein_column)
+
+    return psms
+
+
+def _parse_psms(txt_file, cols, skip_line, log=True):
+    """Parse a single Crux tab-delimited file
+
+    Parameters
+    ----------
+    txt_file : str
+        The comet tab-delimited file to read.
+    cols : list of str
+        The columns to parse.
+    skip_line : bool
+        If true, then skip first row. If false, then read first row.
+
+    Returns
+    -------
+    pandas.DataFrame
+        A :py:class:`pandas.DataFrame` containing the parsed PSMs
+    """
+    if log:
+        LOGGER.info("Reading PSMs from %s...", txt_file)
+
+    if skip_line:
+        return pd.read_csv(
+            txt_file, sep="\t", skiprows=1, usecols=lambda c: c in cols
```

Comment:
The Crux one (which Donovan made) use a lambda instead of a list and I wanted to stay consistent across search engines. I can change everything to use a list instead if you think that is better.